### DISCUSSION TITLE
Getting 405 Method Not Allowed when deployed

### DISCUSSION THREAD

**Discussion by reverbePrintemps** [vercel/next.js#51513] [Help] [answered]
### Summary

Hi,

I have a very simple Next.js (13.4.4-canary.11 - App router) with a single api route at `/api/chat` and a single POST function.

I use Vercel's ai/react package's hook `useChat` which hits the `/api/chat` endpoint by default.

Locally this all works well and dandy but when deployed (https://rpgpt-dun.vercel.app/) I get a `405 Method Not Allowed` error. 

I've looked through a bunch of StackOverflow tickets, as well as GitHub issues and discussions from the Vercel community but nothing stood out.

Any clues would be much appreciated. 

To reproduce: 

1. Visit https://rpgpt-dun.vercel.app/
2. Enter anything in the input box and hit enter
3. Witness the 405 error in the browser Network tab

### Additional information

```js
/api/chat/route.ts
import { Configuration, OpenAIApi } from "openai-edge";
import { OpenAIStream, StreamingTextResponse } from "ai";

const config = new Configuration({
  apiKey: process.env.OPENAI_API_KEY,
});
const openai = new OpenAIApi(config);

// !IMPORTANT
export const runtime = "edge";

export async function POST(req: Request) {
  const { messages } = await req.json();

  const response = await openai.createChatCompletion({
    model: "gpt-3.5-turbo",
    stream: true,
    messages,
  });

  const stream = OpenAIStream(response);
  return new StreamingTextResponse(stream);
}

```
```


### Example

https://rpgpt-dun.vercel.app

**Comment by icyJoseph** [comment]
Are you sure openai can work in the Edge environment?

- https://github.com/openai/openai-node/issues/112
- https://github.com/dan-kwiat/openai-edge not an official package

**Reply by reverbePrintemps** [reply]
This app is basically a copy/paste template from [Vercel themselves](https://github.com/vercel-labs/ai/blob/main/examples/next-openai/app/api/chat/route.ts) and they use the `openai-edge` package so I would assume it works. I just checked the logs for the deployments and I'm seeing `[POST] /api reason=EDGE_FUNCTION_INVOCATION_FAILED, status=500, user_error=true` so there's definitely something wrong with my route handler. Not sure what yet but I'll keep digging. 👍

**Reply by icyJoseph** [reply]
Aha darn line breaks in Mobile... I had missed that you used the edge version already...

**Reply by icyJoseph** [reply]
Could you try catch and log the reason in the catch block? That probably contains an exact reason.

**Reply by icyJoseph** [reply]
One thing I notice is that your app points to `/api` but you have `/api/chat`.

**Reply by reverbePrintemps** [reply]
Actually the useChat() hook points to /api/chat by default. I'll slap in a try catch block in a sec, see what it says.

**Reply by reverbePrintemps** [reply]
Getting `Failed to convert the response to stream. Received status code: 401.` Authorization error? Whaat?

**Reply by icyJoseph** [reply]
Aha, does your key even work? I remember they had something that you ought to pay? The free tier was only available up to 1st of april?

**Reply by reverbePrintemps** [reply]
My key absolutely does work, been paying for it for months now. I'm puzzled...

**Reply by icyJoseph** [reply]
Have you provided the API key? Like in the Vercel dashboard?

**Reply by reverbePrintemps** [reply]
Yup!

**Reply by reverbePrintemps** [reply, marked as answer]
OMG, I had diverging names for the key between my project and the Vercel dashboard 🤦🤦🤦🤦🤦🤦 Thank you for leading me into looking in the right place! I am forever in your debt! Will be closing this discussion and hopefully avoiding further shame. Thanks again! 🍻

**Reply by icyJoseph** [reply]
Ajj it happens 😢 - don't forget to mark as answered 😅
